akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/AllPersistenceIdsStage.scala

```diff
@@ -4,31 +4,40 @@
 
 package akka.persistence.query.journal.leveldb
 
+import akka.NotUsed
 import akka.actor.ActorRef
 import akka.annotation.InternalApi
 import akka.persistence.Persistence
 import akka.persistence.journal.leveldb.LeveldbJournal
-import akka.stream.{ ActorMaterializer, Attributes, Outlet, SourceShape }
-import akka.stream.stage.{ GraphStage, GraphStageLogic, OutHandler, TimerGraphStageLogicWithLogging }
+import akka.stream.Attributes
+import akka.stream.Materializer
+import akka.stream.Outlet
+import akka.stream.SourceShape
+import akka.stream.stage.GraphStage
+import akka.stream.stage.GraphStageLogic
+import akka.stream.stage.OutHandler
+import akka.stream.stage.TimerGraphStageLogicWithLogging
 
 /**
  * INTERNAL API
  */
 @InternalApi
-final private[akka] class AllPersistenceIdsStage(
-    liveQuery: Boolean,
-    writeJournalPluginId: String,
-    mat: ActorMaterializer)
+final private[akka] class AllPersistenceIdsStage(liveQuery: Boolean, writeJournalPluginId: String)
     extends GraphStage[SourceShape[String]] {
 
   val out: Outlet[String] = Outlet("AllPersistenceIds.out")
 
   override def shape: SourceShape[String] = SourceShape(out)
 
-  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = {
-    new TimerGraphStageLogicWithLogging(shape) with OutHandler with Buffer[String] {
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    throw new UnsupportedOperationException("Not used")
```

Comment:
Should the message be more informative?

Reply:
Nevermind, `private[akka] class AllPersistenceIdsStage`

Reply:
and the method will never be called it's just a clunky API to keep backwards compatibility.